homeassistant/components/tolo/number.py

```diff
@@ -0,0 +1,111 @@
+"""TOLO Sauna number controls."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from tololib import ToloClient
+from tololib.message_info import SettingsInfo
+
+from homeassistant.components.number import NumberEntity, NumberEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TIME_MINUTES
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ToloSaunaCoordinatorEntity, ToloSaunaUpdateCoordinator
+from .const import DOMAIN, FAN_TIMER_MAX, POWER_TIMER_MAX, SALT_BATH_TIMER_MAX
+
+
+@dataclass
+class ToloNumberEntityDescriptionBase:
+    """Required values when describing TOLO Number entities."""
+
+    getter: Callable[[SettingsInfo], int | None]
+    setter: Callable[[ToloClient, int | None], Any]
+
+
+@dataclass
+class ToloNumberEntityDescription(
+    NumberEntityDescription, ToloNumberEntityDescriptionBase
+):
+    """Class describing TOLO Number entities."""
+
+    entity_category = EntityCategory.CONFIG
+    min_value = 0
+    step = 1
+
+
+NUMBERS = (
+    ToloNumberEntityDescription(
+        key="power_timer",
+        icon="mdi:power-settings",
+        name="Power Timer",
+        unit_of_measurement=TIME_MINUTES,
+        max_value=POWER_TIMER_MAX,
+        getter=lambda settings: settings.power_timer,
+        setter=lambda client, value: client.set_power_timer(value),
+    ),
+    ToloNumberEntityDescription(
+        key="salt_bath_timer",
+        icon="mdi:shaker-outline",
+        name="Salt Bath Timer",
+        unit_of_measurement=TIME_MINUTES,
+        max_value=SALT_BATH_TIMER_MAX,
+        getter=lambda settings: settings.salt_bath_timer,
+        setter=lambda client, value: client.set_salt_bath_timer(value),
+    ),
+    ToloNumberEntityDescription(
+        key="fan_timer",
+        icon="mdi:fan-auto",
+        name="Fan Timer",
+        unit_of_measurement=TIME_MINUTES,
+        max_value=FAN_TIMER_MAX,
+        getter=lambda settings: settings.fan_timer,
+        setter=lambda client, value: client.set_fan_timer(value),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up number controls for TOLO Sauna."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        [ToloNumberEntity(coordinator, entry, description) for description in NUMBERS]
+    )
```

Comment:
`async_add_entities` accepts a generator :)

```suggestion
    async_add_entities(
        ToloNumberEntity(coordinator, entry, description) for description in NUMBERS
    )
```